python/tank_vendor/shotgun_authentication/interactive_authentication.py

```diff
@@ -330,50 +349,59 @@ def _process_ui():
 
 # Lock the assures only one thread at a time can execute the authentication logic.
 _renew_session_internal_lock = threading.Lock()
+
+class AuthState:
```

Comment:
On a side note, I think we should split the console and the UI based auth into two files at this point. There are quite a few many classes and stuff in this file now!